gcp_variant_transforms/vcf_to_bq.py

```diff
@@ -157,13 +165,53 @@ def _read_variants(pipeline, known_args):
   return variants
 
 
-def _use_beam_header_merger(known_args):
-  if known_args.representative_header_file:
-    return False
+def _get_pipeline_mode(known_args):
+  """Returns the mode the pipeline should operate in based on input size."""
+  if known_args.optimize_for_large_inputs:
+    return PipelineModes.LARGE
+
   match_results = FileSystems.match([known_args.input_pattern])
   if not match_results:
-    return False
-  return len(match_results[0].metadata_list) > _USE_HEADER_MERGER_THRESHOLD
+    raise ValueError('No files matched input_pattern: {}'.format(
+        known_args.input_pattern))
+
+  total_files = len(match_results[0].metadata_list)
+  if total_files > _USE_BEAM_READ_ALL_THRESHOLD:
+    return PipelineModes.LARGE
+  elif total_files > _USE_BEAM_READ_THRESHOLD:
+    return PipelineModes.MEDIUM
+
+  return PipelineModes.SMALL
+
+
+def _merge_headers(known_args, pipeline_args, pipeline_mode):
+  """Merges VCF headers using beam based on pipeline_mode."""
+  if (known_args.representative_header_file
```

Comment:
nit: we've been putting 'or' , 'and' etc on the previous line as a convention so far.

Reply:
Done.